### PR TITLE
Utilize building footprints

### LINKED ISSUE #2: Enable use of existing building footprints

### LINKED ISSUE #34: Buildings that get classified as different damage levels create multiple polygons
As the classification is pixel-wise, portions of buildings may be classified differently. This results in multiple polygons being returned.

### LINKED ISSUE #41: Add Four-GPU support
**Is your feature request related to a problem? Please describe.**
Nope. GPU partitioning should simply be applied for four GPU systems

**Describe the solution you'd like**
Partition GPUs on four GPU systems.

**Describe alternatives you've considered**
Using DP mode exclusively. See #40.

### LINKED ISSUE #42: Force DP mode for single GPU machines
**Is your feature request related to a problem? Please describe.**
One GPU machines fail if not passed dp_mode argument.

**Describe the solution you'd like**
If running inference on single GPU machine, force dp_mode

**Describe alternatives you've considered**
DP mode exclusively. See #40.

### PR DESCRIPTION
**Pull request by pounde**
Closes #2
Closes #34 
Closes #41
Closes #42  
Enables use of building footprint polygons to bypass loc inference. Also enables building damage weighting when input footprints are utilized.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,104 +40,10 @@ On 2 GPUs:
...
 
-# References
- - Competition and Dataset: https://www.xview2.org
- - UNet: https://arxiv.org/pdf/1505.04597.pdf
- - Pretrained models for Pytorch: https://github.com/Cadene/pretrained-models.pytorch
- - My 1st place solution from "SpaceNet 4: Off-Nadir Building Footprint Detection Challenge" (some ideas came from here): https://github.com/SpaceNetChallenge/SpaceNet_Off_Nadir_Solutions/tree/master/cannab
+# Sources
+**xView2 1st place solution**
+  - Model weights from 1st place solution for "xView2: Assess Building Damage" challenge. https://www.xview2.org
+  - More information from original submission see commit: 3fe4a7327f1a19b8c516e0b0930c38c29ac3662b
```
Comment (RitwikGupta) [inline README.md:49]: See `https://github.com/DIUx-xView/xView2_first_place` instead

File: handler.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 import rasterio.warp
 import rasterio.crs
 import torch
-#import ray
+# import ray
```
Comment (RitwikGupta) [inline handler.py:17]: You can just delete this line

File: handler.py [hunk 3]
```diff
@@ -129,33 +138,36 @@ def postprocess_and_write(result_dict):
...
+        loc_preds = []
+        _i = -1
+        for k, v in result_dict.items():
+            if 'loc' in k:
+                _i += 1
+                msk = v['loc'].numpy()
+                loc_preds.append(msk * loc_coefs[_i])
+
+        loc_preds = np.asarray(loc_preds).astype('float').sum(axis=0) / np.sum(loc_coefs) / 255
+        msk_loc = (1 * ((loc_preds > _thr[0]) | ((loc_preds > _thr[1]) & (msk_dmg > 1) & (msk_dmg < 4)) | (
```
Comment (RitwikGupta) [inline handler.py:166]: This code is so hairy. Can we add in some comments for documentation?

**Review by RitwikGupta — COMMENTED** [review]
Minor documentation changes needed

**Comment by pounde** [comment]
> This code is so hairy. Can we add in some comments for documentation? 

Nope...well, maybe you or Jared can. haha

File: handler.py (hunk 3 above)
Comment (pounde) [reply handler.py:166]: Nope...well, maybe you or @jdunnmon can.

**Comment by pounde** [comment]
Changes made, save one (handler.py lines 159-166). All files formatted with black.

**Pull request merged**